File: Algorithms/Prefix Sum/prefix_sum.java
public class PrefixSum {

    private static int[] prefixSum = null;

    public static void calculatePrefixSum(int[] arr) {
        prefixSum = new int[arr.length + 1];
        for (int k = 1; k <= arr.length; k++) {
            prefixSum[k] = prefixSum[k - 1] + arr[k - 1];
        }
    }

    public static int sumRange(int[] arr, int i, int j) {
        if (prefixSum == null) {
            calculatePrefixSum(arr);
        }
        return prefixSum[j + 1] - prefixSum[i];
    }
}

File: Algorithms/Prefix Sum/prefix_sum.py
prefix_sum = None

def calculate_prefix_sum(arr):
    global prefix_sum
    prefix_sum = [0] * (len(arr) + 1)
    for k in range(1, len(arr) + 1):
        prefix_sum[k] = prefix_sum[k - 1] + arr[k - 1]

def sum_range(arr, i, j):
    global prefix_sum
    if prefix_sum is None:
        calculate_prefix_sum(arr)
    return prefix_sum[j + 1] - prefix_sum[i]
